profiler/src/ProfilerEngine/Datadog.Profiler.Native/ClrEventsParser.cpp

```diff
@@ -319,6 +319,10 @@ void ClrEventsParser::NotifyGarbageCollectionStarted(uint64_t timestamp, int32_t
 {
     for (auto& pGarbageCollectionsListener : _pGarbageCollectionsListeners)
     {
+        std::stringstream buffer;
+        buffer << "OnGarbageCollectionStart: " << number << " " << generation << " " << reason << " " << type;
```

Comment:
[PERF] I expect this log to be written in Debug only, so I would expect the `std::stringstream` to be construction only if we are debug level no?